source/github/github.go

```diff
@@ -20,6 +20,8 @@ func init() {
 	source.Register("github", &Github{})
 }
 
+const DefaultClientAuth = "u:p"
```

Comment:
It's very unusual to have a client named 'u' having password 'p' at the same time. So it's unlikely to change an authenticated client to unauthenticated client unintentionally.

Reply:
Instead of having this special/sentinel value, support the user/password not being specified. Change line 53/54 below:
```golang
if u.User == nil {
	return nil, ErrNoUserInfo
}
```